### PR TITLE
feat(api): Add Ed25519 sign/verify SubtleCrypto algo

### PR DESCRIPTION
**Pull request by bsmth**
#### Summary

Currently [the spec states](https://w3c.github.io/webcrypto/#scope-algorithms):

>Because the underlying cryptographic implementations will vary between conforming user agents... this specification does not dictate a mandatory set of algorithms that MUST be implemented. Instead, it defines a common set of bindings that can be used in an algorithm-independent manner...

There are a list of recommendations for [implementers](https://www.w3.org/TR/WebCryptoAPI/#algorithm-recommendations-implementers):

>In order to promote interoperability for developers, this specification includes a list of suggested algorithms. These are considered to be the most widely used algorithms in practice at the time of writing, and therefore provide a good starting point for initial implementations of this specification. The suggested algorithms are:
>
>[HMAC](https://www.w3.org/TR/WebCryptoAPI/#hmac) using [SHA-1](https://www.w3.org/TR/WebCryptoAPI/#alg-sha-1)
[HMAC](https://www.w3.org/TR/WebCryptoAPI/#hmac) using [SHA-256](https://www.w3.org/TR/WebCryptoAPI/#alg-sha-256)
[RSASSA-PKCS1-v1_5](https://www.w3.org/TR/WebCryptoAPI/#rsassa-pkcs1) using [SHA-1](https://www.w3.org/TR/WebCryptoAPI/#alg-sha-256)
[RSA-PSS](https://www.w3.org/TR/WebCryptoAPI/#rsa-pss) using [SHA-256](https://www.w3.org/TR/WebCryptoAPI/#alg-sha-256) and MGF1 with [SHA-256](https://www.w3.org/TR/WebCryptoAPI/#alg-sha-256).
[RSA-OAEP](https://www.w3.org/TR/WebCryptoAPI/#rsa-oaep) using [SHA-256](https://www.w3.org/TR/WebCryptoAPI/#alg-sha-256) and MGF1 with [SHA-256](https://www.w3.org/TR/WebCryptoAPI/#alg-sha-256).
[ECDSA](https://www.w3.org/TR/WebCryptoAPI/#ecdsa) using [P-256](https://www.w3.org/TR/WebCryptoAPI/#dfn-NamedCurve-p256) curve and [SHA-256](https://www.w3.org/TR/WebCryptoAPI/#alg-sha-256)
[AES-CBC](https://www.w3.org/TR/WebCryptoAPI/#aes-cbc)

I've added a sub-feature for an algo which is not part of this recommendation, but is available in multiple engines.

Standardization status:

> There's a WICG draft for Curve25519 and Curve448 in [Web Crypto here](https://wicg.github.io/webcrypto-secure-curves/)

#### Test results and supporting details

* **Chrome:** 113 (behind flag)
   - Commit https://chromium-review.googlesource.com/c/chromium/src/+/4016576
   - Commit in find release tool: https://storage.googleapis.com/chromium-find-releases-static/2db.html#2db4c847d07e2811f9efff4387f20906d109dedf
   - Flag: `enable-experimental-web-platform-features` (chrome://flags/#enable-experimental-web-platform-features)

* **Safari:** Stable v17
  * Relnote https://webkit.org/blog/13839/release-notes-for-safari-technology-preview-163/
  * Commit https://github.com/WebKit/WebKit/commit/3abe0e45310ef3e846f8509e44b5514fb2099f76 

* **Node.js:** 16.17.0 (marked as experimental)
  * Docs - https://nodejs.org/dist/latest-v16.x/docs/api/webcrypto.html#ed25519ed448x25519x448-key-pairs  
  * Changelog https://github.com/nodejs/node/commit/5e5fb825fc4bc3cd3e51614aaad9af14342a5e6c

* **Deno:** 1.26.0 
  * Changelog https://github.com/denoland/deno/blob/main/Releases.md#1260--20220928

* No Fx support

* **WPT:** generateKey
  * https://wpt.live/WebCryptoAPI/generateKey/successes_Ed25519.https.any.html 
  * https://wpt.fyi/results/WebCryptoAPI/generateKey/successes_Ed25519.https.any.html?label=master&product=chrome%5Bexperimental%5D&product=deno%5Bexperimental%5D&product=firefox%5Bexperimental%5D&product=safari%5Bexperimental%5D&product=node.js%5Bexperimental%5D&product=chrome%5Bstable%5D&product=safari%5Bstable%5D&product=deno%5Bstable%5D&product=node.js%5Bstable%5D&aligned

Node repro:

```js
async function generateAndSignKey() {
  const algorithm = { name: "Ed25519" };
  const encoder = new TextEncoder();
  const cryptoKey = await crypto.subtle.generateKey(
    algorithm,
    true,
    ["sign", "verify"]
  );

  console.log(
    await crypto.subtle.sign(
      algorithm,
      cryptoKey.privateKey,
      new Uint8Array([255])
    )
  );
}

generateAndSignKey();
```

output:

```
(node:27247) ExperimentalWarning: The Ed25519 Web Crypto API algorithm is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
ArrayBuffer {
  [Uint8Contents]: <bd 47 7d 05...
//
node --version
v20.11.0
```

#### Related issues

* https://github.com/WICG/webcrypto-secure-curves/issues/20
* https://github.com/mdn/content/issues/30886
* https://github.com/mdn/dom-examples/pull/247

### CODE REVIEW

**Comment by guest271314** [comment]
Bun also supports Ed25519 algorithm via `import { webcrypto } from "node:crypto".`

**Comment by bsmth** [comment]
> Bun also supports Ed25519 algorithm via `import { webcrypto } from "node:crypto".`

Bun is not (yet?) included in Browsers/Engines here. See https://github.com/mdn/browser-compat-data/issues/18484

**Comment by guest271314** [comment]
I'm talking about technical capabilities, not bureaucracy.

**Comment by bsmth** [comment]
> I'm talking about technical capabilities, not bureaucracy.

I want to remind you that everyone collaborating in Mozilla spaces, including online spaces like these GitHub projects, must adhere to [Mozilla Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/) (CPG). Being rude, dismissive, or disrespectful will not be tolerated. Please familiarize yourself with the guidelines for CPG and the [Open source etiquette](https://developer.mozilla.org/en-US/docs/MDN/Community/Open_source_etiquette) guide. Thank you.

**Comment by guest271314** [comment]
@bsmth  

The post I replied to listed JavaScript runtimes that support Web Cryptography API. 

I stated Bun supports Ed25519.

You linked to an issue about linking Bun in BCD, which is a purely bureaucratic, politic discussion and decision by MDN Web Docs/Mozilla that has absolutely nothing to do with the technical capabilities of the runtime itself.

I don;t know why you linked to CoC and such as if the above is not a fact.

If you never heard of Bun, look it up, perform some due diligence, download and test the technical capabilities of the runtime.

File: api/SubtleCrypto.json [hunk 1]
```diff
@@ -670,6 +670,55 @@
             "standard_track": true,
             "deprecated": false
           }
+        },
+        "ed25519": {
+          "__compat": {
+            "description": "<code>Ed25519</code> algorithm",
+            "mdn_url": "https://developer.mozilla.org/docs/Web/API/SubtleCrypto/sign",
```
Comment (Elchi3) [inline api/SubtleCrypto.json:677]: This is a sub-feature of exportKey, why link to sign?

File: api/SubtleCrypto.json [hunk 2]
```diff
@@ -670,6 +670,55 @@
...
+            "spec_url": "https://w3c.github.io/webcrypto/#SubtleCrypto-method-sign",
```
Comment (Elchi3) [inline api/SubtleCrypto.json:678]: Wouldn't it be better to link to https://wicg.github.io/webcrypto-secure-curves/#ed25519?

File: api/SubtleCrypto.json [hunk 3]
```diff
@@ -670,6 +670,55 @@
...
+              "firefox": {
+                "version_added": false
+              },
+              "firefox_android": "mirror",
+              "ie": {
+                "version_added": false
+              },
+              "nodejs": {
+                "version_added": "16.17.0",
+                "notes": "Marked as ['Stability 1' - Experimental](https://nodejs.org/dist/latest-v16.x/docs/api/webcrypto.html#ed25519ed448x25519x448-key-pairs)"
```
Comment (Elchi3) [inline api/SubtleCrypto.json:704]: BCD doesn't support markdown yet. Please change to HTML for the link

**Review by Elchi3 — CHANGES_REQUESTED** [review]
Thanks @bsmth, some inline comments for you to look at. They apply to all for additions; I only commented on the first one.

**Comment by bsmth** [comment]
Thanks a lot, will take a look at your comments shortly 👍🏻

File: api/SubtleCrypto.json (hunk 1 above)
Comment (bsmth) [reply api/SubtleCrypto.json:677]: copy-paste err, thanks for spotting

**Review by Elchi3 — APPROVED** [review]
Thanks @bsmth, looks great to me! 👍

**Comment by bsmth** [comment]
Thank you! 🙏🏻

**Pull request merged**